### PR TITLE
Minor fixes to migrate-if-needed script

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref #22448 #20504

@lavalamp

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34438)

### CODE REVIEW

**Comment by wojtek-t** [comment]
@mml - friendly ping

File: cluster/images/etcd/migrate-if-needed.sh [hunk 1]
```diff
@@ -73,6 +73,18 @@ if [ -e "${DATA_DIRECTORY}/${VERSION_FILE}" ]; then
   CURRENT_STORAGE="$(echo $VERSION_CONTENTS | cut -d '/' -f 2)"
 fi
 
+# If there is no data in DATA_DIRECTORY, this means that we are
+# starting etcd from scratch. In that case, we don't need to do
+# any migration.
+if [ ! -d "${DATA_DIRECTORY}" ]; then
+  mkdir -p "${DATA_DIRECTORY}"
+fi
+if [ ! "$(ls -A ${DATA_DIRECTORY})" ]; then
```
Comment (mml) [inline cluster/images/etcd/migrate-if-needed.sh]: I don't think we want `[ !` here... I think we want `[ -z`.

**Review by mml — CHANGES_REQUESTED** [review]
LG otherwise.  Feel free to apply label after you fix the test.

**Comment by mml** [comment]
@k8s-bot gci gce e2e test this issue: #33388

**Comment by wojtek-t** [comment]
Comment applied, adding lgtm label based on above comment.

**Pull request merged**